src/main/java/roomescape/reservation/ReservationService.java

```diff
@@ -12,13 +15,25 @@ public ReservationService(ReservationDao reservationDao) {
         this.reservationDao = reservationDao;
     }
 
-    public ReservationResponse save(ReservationRequest reservationRequest) {
+    public ReservationResponse save(ReservationRequest reservationRequest, LoginMember loginMember) {
+        if (!reservationRequest.getName().equals(loginMember.name()) && !loginMember.isAdmin()) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN, "본인 이름으로만 예약할 수 있습니다.");
+        }
```

Comment:
조건문이 not and not이면 코드 가독성이 떨어집니다
도메인에 메서드를 만들어서 조금 개선해보면 어떨까요?

아니면 admin용 service & controller와 일반 유저용 service & controller를 분리하는 방법도 있어요!

Reply:
```suggestion
        if (loginMember.doesNotHaveName(reservationRequest.getName()) && loginMember.isNotAdmin()) {
            throw new ResponseStatusException(HttpStatus.FORBIDDEN, "본인 이름으로만 예약할 수 있습니다.");
        }
```

근데 `doesNotHaveName` 메서드명이 맘에 안들긴 하네요 ㄷㄷ

Reply:
그리고 403을 반환하는 방식으로 코드를 작성하신 것 같은데,
클라이언트는 400을 응답받고 예외 메시지도 받지 못하고 있어요!